src/hubbleds/story.py

```diff
@@ -400,7 +401,44 @@ def update_student_data(self, *args):
                                                  color="blue",
                                                  alpha=1,
                                                  markersize=10)
+    
+    def update_example_galaxy_data(self, *args):
+        print('update_example_galaxy_data')
+        dc = self.data_collection
+        meas_data = dc[EXAMPLE_GALAXY_MEASUREMENTS]
+        df = meas_data.to_dataframe()
+
+        df = df[df[DISTANCE_COMPONENT].notna() & \
+                df[VELOCITY_COMPONENT].notna() & \
+                df[ANGULAR_SIZE_COMPONENT].notna()]
+        df[NAME_COMPONENT] = df[NAME_COMPONENT].astype(np.dtype(str))
+        main_components = [x.label for x in meas_data.main_components]
+        components = { col: list(df[col]) for col in main_components }
+
+        if not all(len(v) > 0 for v in components.values()):
+            return
+        
+        nrows = [len(v) for v in components.values()][0]
+        
+        new_data = {} #Data(label=EXAMPLE_GALAXY_MEASUREMENTS)
+        for col, data in components.items():
+            categorical = col in self.categorical_cols
+            ctype = CategoricalComponent if categorical else Component
+            comp = ctype(np.array(data))
+            new_data.update({comp: np.array(data)})
+        
+        example_galaxy_data = dc[EXAMPLE_GALAXY_MEASUREMENTS]
+        example_galaxy_data.update_values_from_data(new_data)
+        HubblesLaw.make_data_writeable(example_galaxy_data)
                 
+    def update_data_values(self, dc_name, values, index):
```

Comment:
I don't see why we need this function - it looks exactly the same as the superclass implementation

Reply:
I thought that was from `Stage` in `phases.py`. Does `HubblesLaw` or `Story` inherit that somehow?

Reply:
Oh oops, I thought this was in stage.py! Feel free to ignore, sorry